File: Python/560_Subarray_Sum_Equals_K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        summ = 0
        d = {}
        d[0] = 1
        for i in range(len(nums)):
            summ += nums[i]
            if summ - k in d:
                count += d[summ-k]
            d[summ] = d.get(summ, 0) + 1
        return count
                

File: C++/560_Subarray_Sum_Equals_K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int begin = 0, end = 0;
        int sum = 0;
        int ans = 0;
        for(int i = 0; i < nums.size(); i++){
            sum = 0;
            begin = i;
            end = i;
            while(end < nums.size()){
                sum += nums[end];
                if(sum == k) {
                    ans++;
                }
                end++;
            }
        }
        return ans;
    }
};